### PR TITLE
refactor(tests): Added new pre-commit hook for ordering functions 

### PR DESCRIPTION
**Pull request by NripeshN**

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Review by AnnaTz — APPROVED** [review]

**Comment by AnnaTz** [comment]
You'll have to resolve the conflicts to sync with the latest main branch, but looks good overall. Thanks!

**Review by sherry30 — COMMENTED** [review]
I would prefer if you leave out helpers folder from the test_ivy folder.
just refactoring the test functions would be nice

**Comment by NripeshN** [comment]
> I would prefer if you leave out helpers folder from the test_ivy folder. just refactoring the test functions would be nice

I have ignored all files in the helpers folder now!

File: ivy_tests/test_ivy/test_frontends/config/jax.py
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_frontends/config/jax.py:1]: I feel this file and other files under `test_frontends/config` should be excluded from this lint

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hey @NripeshN,
I only have this comment that I listed below, other than that LGTM

**Pull request merged**